src/index.ts

```diff
@@ -1210,10 +1265,10 @@ export class ValidationService {
     /**
      * Scans the provided root element for any validation directives and attaches behavior to them.
      */
-    scan(root: ParentNode) {
+    scan(root: ParentNode, remove: boolean = false) {
```

Comment:
> Not sure if it's cleaner to use the `removed` boolean that's currently being passed to `scan` and its children or to have a separate set of functions instead

IMO the `removed` boolean is fine for the private functions, but not ideal for a public method. Suggest reverting the `scan()` change and adding a `remove(root: ParentNode)` that calls the helpers with `remove = true`.

That said, I don't disagree with your intuition that `remove` isn't particularly clean. You'd have to be careful with `this`, but `scanMessages`/`scanInputs` could receive functions to call in their loop(s):

```diff
     scan(root: ParentNode) {
         this.logger.log('Scanning', root);
-        this.scanMessages(root);
-        this.scanInputs(root);
+        this.scanMessages(root, this.pushValidationMessageSpan);
+        this.scanInputs(root, this.addInput);
     }
```

Reply:
OK, I added a separate `remove` function and left the private functions as is for now.